compiler/src/dotty/tools/dotc/ast/NavigateAST.scala

```diff
@@ -74,9 +74,23 @@ object NavigateAST {
       }
       path
     }
-    def singlePath(p: Positioned, path: List[Positioned]): List[Positioned] =
-      if (p.pos contains pos) childPath(p.productIterator, p :: path)
-      else path
+    def singlePath(p: Positioned, path: List[Positioned]): List[Positioned] = {
+      val currentHasPos =
+        p.pos.contains(pos) && (!p.pos.isZeroExtent ||
+          (path match {
+            case p1 :: _ => p1.pos.isZeroExtent || !p1.pos.contains(pos)
```

Comment:
Indeed, I forgot to clean this up, and trying to document it I realized it's broken: I was trying to get the next node in the tree that the path could go through, but this is actually pretty hard to do. I've given up on that and instead added a `skipZeroExtent` parameter in `NavigateAST#pathTo` (that I use in `Interactive#pathTo`): https://github.com/lampepfl/dotty/pull/2532/commits/b3befee6f4b9ce09babfc577c39fdbd6d2b4f299